pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/AnswerDao.java

```diff
@@ -291,6 +292,9 @@ default void deleteAnswers(Set<Long> answerIds) {
             }
         }
 
+        // Child answers are deleted separately, so remove any potential ones from the set of top-level answer ids.
+        answerIds.removeAll(childAnswerIds);
```

Comment:
This was a latent bug. Didn't run into issues here until I have added the new test on the composite answers.